test/dlc_tests/sanity/test_security_check.py

```diff
@@ -74,3 +80,38 @@ def test_ecr_scan(image, ecr_client):
         f"Found vulnerabilities in image {image}: {str(severity_counts)}\n"
         f"Vulnerabilities: {json.dumps(scan_results, indent=4)}"
     )
+
+
+def _reupload_image_to_test_ecr(source_image_uri, test_ecr_client, test_region, test_account_id):
+    """
+    Helper function to reupload an image owned by a different account to an ECR repo in this account, so that
+    this account can freely run ECR Scan without permission issues.
+
+    :param source_image_uri: str Image URI for image to be tested
+    :param test_ecr_client: boto3.Client ECR client for account where test is being run
+    :param test_region: str Region where test is being run
+    :param test_account_id: str Account ID for account where test is being run
+    """
+    ctx = Context()
+    image_account_id = get_account_id_from_image_uri(source_image_uri)
+    image_region = get_region_from_image_uri(source_image_uri)
+    login_to_ecr_registry(ctx, image_account_id, image_region)
+    ctx.run(f"docker pull {source_image_uri}")
+
+    image_repo_uri, image_tag = source_image_uri.split(":")
+    _, image_repo_name = image_repo_uri.split("/")
+    test_image_repo_name = f"beta-{image_repo_name}"
+    if not ecr_utils.ecr_repo_exists(test_ecr_client, test_image_repo_name):
+        ecr_utils.create_ecr_repository(test_ecr_client, test_image_repo_name)
```

Comment:
Seems a bit unsafe to create repos on the fly. Plus as you said, jobs don't even have such permissions. Maybe just throw an exception here?

Reply:
Sure, can do that.